File: android/src/main/java/expo/modules/nvm/NvmModule.kt
package expo.modules.nvm

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import com.facebook.react.bridge.ReadableMap
import expo.modules.kotlin.Promise

class NvmModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("Nvm")

    AsyncFunction("hello") { params: ReadableMap, promise: Promise ->
      promise.resolve(params.toString())
    }
  }
}

File: ios/NvmModule.swift
import ExpoModulesCore

public class NvmModule: Module {
  public func definition() -> ModuleDefinition {
    Name("Nvm")

    AsyncFunction("hello") { (_ params: [String: Any], promise: Promise) in
      print(params)
      promise.resolve(
        (params.flatMap({ (key, value) -> String in
          return "\(key)=\(value)"
        }) as Array).joined(separator: ";")
      )
    }.runOnQueue(.main)
  }
}
